pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -36,47 +38,68 @@ import (
 type topologyPairToScore map[string]map[string]int64
 
 type podAffinityPriorityMap struct {
-	// nodes contain all nodes that should be considered.
-	nodes []*v1.Node
-	// tracks a topology pair score so far.
-	topologyScore topologyPairToScore
+	topologyScore         topologyPairToScore
+	pod                   *v1.Pod
+	affinityTerms         []*processedWeightedAffinityTerm
+	antiAffinityTerms     []*processedWeightedAffinityTerm
+	sharedLister          schedulerlisters.SharedLister
+	hardPodAffinityWeight int32
 	sync.Mutex
 }
 
-func newPodAffinityPriorityMap(nodes []*v1.Node) *podAffinityPriorityMap {
-	return &podAffinityPriorityMap{
-		nodes:         nodes,
-		topologyScore: make(topologyPairToScore),
-	}
+type processedWeightedAffinityTerm struct {
+	namespaces  sets.String
+	selector    labels.Selector
+	weight      int32
+	topologyKey string
 }
 
-func (p *podAffinityPriorityMap) processTerm(term *v1.PodAffinityTerm, podDefiningAffinityTerm, podToCheck *v1.Pod, fixedNode *v1.Node, weight int64) error {
-	namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(podDefiningAffinityTerm, term)
+func getProcessedWeightedAffinityTerm(pod *v1.Pod, term *v1.PodAffinityTerm, weight int32) (*processedWeightedAffinityTerm, error) {
+	namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(pod, term)
 	selector, err := metav1.LabelSelectorAsSelector(term.LabelSelector)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	return &processedWeightedAffinityTerm{namespaces: namespaces, selector: selector, topologyKey: term.TopologyKey, weight: weight}, nil
+}
+
+func getProcessedTerms(pod *v1.Pod, terms []v1.WeightedPodAffinityTerm) ([]*processedWeightedAffinityTerm, error) {
+	if terms == nil {
+		return nil, nil
 	}
+
+	var processedTerms []*processedWeightedAffinityTerm
+	for i := range terms {
+		p, err := getProcessedWeightedAffinityTerm(pod, &terms[i].PodAffinityTerm, terms[i].Weight)
+		if err != nil {
+			return nil, err
+		}
+		processedTerms = append(processedTerms, p)
+	}
+	return processedTerms, nil
+}
+
+func (p *podAffinityPriorityMap) processTerm(term *processedWeightedAffinityTerm, podDefiningAffinityTerm, podToCheck *v1.Pod, fixedNode *v1.Node, weight int64) error {
```

Comment:
podDefiningAffinityTerm is unused :)

Reply:
cool, that is because of the refactoring.